airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -189,7 +213,8 @@ def read_records(
         and update `start_date` key in the `stream_slice` with the latest read record's cursor value, then retry the sync.
         """
         while True:
-            self.logger.info("Starting a while loop iteration")
+            self.incremental_sieve_logger.bump()
```

Comment:
Going off the prior comment about separate slice and read_records log cycles, do you think it's also worth initializing a new sieve every time read_records is invoked? If the sieve is shared on the stream, we will emit exactly 1/10 of logs, but it could be a little confusing why certain cycles are skipped on different read_records invocations if a prior one has a partially bumped count.

I'm open to either option and this is non-blocking on approval, but I think worth highlighting when the sieve is shared at this point? If we do decide to have a separate sieve per `read_records` invocation, we could probably remove it from `__init__()` and do something similar to slices where it is created within this method.

Reply:
That's a tricky point cause normally a while loop here contains a single iteration. It only goes further if there were errors during the read. So I do not think there's much sense in initializing new sieve each time `read_records` is invoked. On the other hand I agree that logs may become inconsistent if there are two or more while loop iterations, therefore I moved `incremental_sieve_logger.bump()` outside the loop to remain consistent